blueocean-dashboard/src/main/js/api/Pipeline.js

```diff
@@ -36,10 +36,17 @@ export default class Pipeline {
         if (runId === undefined) {
             throw new Error('Branch.runDetailsRouteUrl must be supplied with a "runId" parameter.');
         }
+
+        // clean the branch name before passing it through to URL builder
+        let branchName = this.branchName;
+        while (branchName.indexOf('%2') >= 0) {
```

Comment:
NIT ... I've seen others do it too ... all docs for `string.indexOf` that've read state something like "_or -1 if it never occurs_", yet people code that into conditions as `>= 0` (as opposed to `!== -1`) 😸